Automattic/wp-calypso
# Nav Unification: Remove old sidebar (part 1)

Part of https://github.com/Automattic/wp-calypso/issues/47051.

#### Changes proposed in this Pull Request

This PR is the first part (more to come!) of a cleanup task which seeks to remove no longer used code as of the Nav Unification launch:
- 949a578252062645587a9149309fd1a898a6863d removes the `MobileBackToSidebar` component since it doesn't render anything when Nav Unification is enabled.
- e7fca17f559710946594285a75f7a05ffdaa56ed removes the `MeSidebarNavigation` component since it doesn't render anything when Nav Unification is enabled.
- 13bbfa0d45eeaec613bd4ef6646dfe191b9ff192 ensures that the `MySitesSidebarNavigation` component only gets rendered on `cloud.jetpack.com` pages.
- 43c0f7c2e4f8d12a03efedfb5ec1c2d9a79cb166 refactors the `SidebarNavigation` component so we no longer need `MySitesSidebarNavigation`.

#### Context

Last year, we introduced new components for the navigation sidebar as part of the Nav Unification project. However, we've been falling back to the old sidebar components in these scenarios:
- As an escape hatch using [the `?disable-nav-unification` query param](https://github.com/Automattic/wp-calypso/blob/c7e06d5e2fe1ca94643d16855209704c39452f30/client/state/selectors/is-nav-unification-enabled.js#L9-L12).
- For self-hosted Jetpack sites [not updated to Jetpack 9.8](https://github.com/Automattic/wp-calypso/blob/c7e06d5e2fe1ca94643d16855209704c39452f30/client/state/selectors/is-nav-unification-enabled.js#L14-L19).
- [In Jetpack Cloud](https://github.com/Automattic/wp-calypso/blob/c7e06d5e2fe1ca94643d16855209704c39452f30/client/my-sites/navigation/index.jsx#L24-L25).

We no longer need to support the first two scenarios, since the escape hatch was meant to be temporary and Calypso does not need to support lower Jetpack versions than 9.8 anymore.

As for the third scenario, Jetpack Cloud only needs a small subset of the old components.

That gives us a good opportunity to remove a lot of unused code which would simplify the overall Calypso codebase.

#### Testing instructions

- Use the Calypso live link below.
- Make sure there are no regressions in the navigation sidebar after a site is selected or when visiting the `me` section.
- Use the Jetpack Cloud live link below.
- Make sure there are no regressions in the navigation sidebar.
- Make sure that we still display a navigation header on mobile viewports that toggles the sidebar: <img width="445" alt="Screen Shot 2022-02-28 at 12 33 16" src="https://user-images.githubusercontent.com/1233880/155976623-cb35d694-7e6d-4a23-a78e-9cf54e8454e1.png">
